src/liger_kernel/ops/cross_entropy.py

```diff
@@ -213,6 +213,8 @@ def cross_entropy_forward(_input, target, ignore_index, label_smoothing, reducti
         target = target.contiguous()
 
     # Here we use a trick to store X_ptr gradient in X_ptr so we can save memory
+    # explicitly declare in-place operation is performed
+    _input.add_(0)
```

Comment:
can you explain a bit on what this is doing exactly?

Reply:
[How](https://pytorch.org/docs/stable/autograd.html#in-place-correctness-checks) torch checks whether an in-place op on a tensor would result in an incorrect gradient calculation, is by implicitly tracking the "version" of a tensor and comparing the saved version and afterward version.

When an in-place operation is performed on a tensor, the version of the tensor is incremented by 1, achieved by an [internal function](https://github.com/pytorch/pytorch/blob/190e09d8b6a13f789b143f0fbd1325f924550967/c10/core/TensorImpl.h#L382) `bump()` written in C.

Since the `bump()` function is only called when doing "torch" in-place operation, i.e. in-place operations in "triton kernel" cannot trigger `bump()`, which makes torch lose track of the version and unable to raise an error.

This approach is a hint for torch, by manually performing a torch's in-place op when we do that tensor dirty in triton kernel.

Reference:
torch in-place-correctness-checks
https://pytorch.org/docs/stable/autograd.html#in-place-correctness-checks
version tracking, bump (there's a note above the function block)
https://github.com/pytorch/pytorch/blob/190e09d8b6a13f789b143f0fbd1325f924550967/c10/core/TensorImpl.h#L382

Reply:
Does this operation have any performance impact? cc @ByronHsu

Reply:
impact is huge for add_(0). will try some other inplace ops.